### DISCUSSION TITLE
Fetching domains by an API

### DISCUSSION THREAD

**Discussion by sorenhl** [nuxt-modules/i18n#1384] [Q&A]
Hi,
We are using nuxt as a part of a larger solution where the language and domains are controlled in another system (CMS).Is it possible to hook into the process and do and resolve the current language by an API request?

**Comment by rchl** [comment]
By "domains" do you meant the `domain` property on locales?

This needs to be defined at build time so you'd have to make an API call in your nuxt.config.js and create module configuration based on that.

**Reply by sorenhl** [reply]
Yes, this is the property. The domain can be solved at runtime if you place them in vuex and use environment variables, which is what I do now. 
But as I have several systems that needs to be aware of the domain. I would therefore prefer to have an asynchronous method that solves the language, i.e. where I can call an endpoint with /api/GetLocaleByDomain?domain=xxx and get the correct locale back.
